Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,371 @@
+                normalizedSplinePosition = m_PreviousNormalizedSplinePosition - offset;
+            }
+            m_PreviousNormalizedSplinePosition = normalizedSplinePosition;
+            m_Spline.Evaluate(normalizedSplinePosition, 
+                out var localPosition, out var localTangent, out var localUp);
+            Vector3 newCameraPos = localPosition;
+            if (Vector3.SqrMagnitude(localTangent) != 0)
+            {
+                m_CorrectedTangent = localTangent;
+            }
+            if (Vector3.SqrMagnitude(localUp) != 0)
+            {
+                m_CorrectedUp = localUp;
+            }
+            var newSplineOrientation = Quaternion.LookRotation(m_CorrectedTangent, m_CorrectedUp);
+
+            var rollExt = rollExtension;
+            if (rollExt != null && rollExt.enabled)
+            {
+                float roll = rollExt.RollOverride.Evaluate(spline, normalizedSplinePosition, 
+                    PathIndexUnit.Normalized, new UnityEngine.Splines.Interpolators.LerpFloat());
+                var rollRotation = Quaternion.AngleAxis(-roll, m_CorrectedTangent);
+                newSplineOrientation = Quaternion.LookRotation(m_CorrectedTangent, rollRotation * m_CorrectedUp);
+            }
+
+            // Apply the offset to get the new camera position
+            var offsetX = newSplineOrientation * Vector3.right;
+            var offsetY = newSplineOrientation * Vector3.up;
+            var offsetZ = newSplineOrientation * Vector3.forward;
+            newCameraPos += m_SplineOffset.x * offsetX;
+            newCameraPos += m_SplineOffset.y * offsetY;
+            newCameraPos += m_SplineOffset.z * offsetZ;
+
+            // Apply damping to the remaining directions
+            if (deltaTime >= 0 && VirtualCamera.PreviousStateIsValid)
+            {
+                Vector3 currentCameraPos = m_PreviousCameraPosition;
+                Vector3 delta = (currentCameraPos - newCameraPos);
+                Vector3 delta1 = Vector3.Dot(delta, offsetY) * offsetY;
+                Vector3 delta0 = delta - delta1;
+                delta0 = Damper.Damp(delta0, m_Damping.x, deltaTime);
+                delta1 = Damper.Damp(delta1, m_Damping.y, deltaTime);
+                newCameraPos = currentCameraPos - (delta0 + delta1);
+            }
+            curState.RawPosition = m_PreviousCameraPosition = newCameraPos;
+
+            // Set the orientation and up
+            Quaternion newOrientation = GetCameraOrientationAtSplinePoint(newSplineOrientation, curState.ReferenceUp);
+            if (deltaTime >= 0 && VirtualCamera.PreviousStateIsValid)
+            {
+                Vector3 relative = (Quaternion.Inverse(m_PreviousOrientation)
+                    * newOrientation).eulerAngles;
+                for (int i = 0; i < 3; ++i)
+                    if (relative[i] > 180)
+                        relative[i] -= 360;
+                relative = Damper.Damp(relative, m_AngularDamping, deltaTime);
```

Comment:
I think you need to do a proper spherical damping here, not a componentwise euler damping.
See implementation in CinemachineTransposer line 270